### PR TITLE
part 1 of 4988 PR from tg.

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

**Comment by eswordthecat** [comment]
Can be pulled if success.

File: code/game/machinery/telecomms/broadcaster.dm [hunk 1]
```diff
@@ -261,7 +261,7 @@ var/message_delay = 0 // To make sure restarting the recentmessages list is kept
 				radios |= R
 
 	// Get a list of mobs who can hear from the radios we collected.
-	var/list/receive = get_movables_in_radio_ranges(radios)
+	var/list/receive = get_mobs_in_radio_ranges(radios)
```
Comment (ComicIronic) [inline code/game/machinery/telecomms/broadcaster.dm]: But this is deliberately movables.

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -225,32 +225,17 @@
...
-	for(var/i = 1; i <= player_list.len; i++)
-		var/mob/M = player_list[i]
-		if(M)
-			var/turf/ear = get_turf(M)
-			if(ear)
-				// Ghostship is magic: Ghosts can hear radio chatter from anywhere
-				if(speaker_coverage[ear] || (istype(M, /mob/dead/observer) && (M.client) && (M.client.prefs.toggles & CHAT_GHOSTRADIO)))
-					. |= M		// Since we're already looping through mobs, why bother using |= ? This only slows things down.
-	return .
+	for (var/obj/item/device/radio/R in radios)
```
Comment (ComicIronic) [inline code/__HELPERS/game.dm]: Index loops are faster than type loops.

Comment (eswordthecat) [inline code/__HELPERS/game.dm]: They're prone to index out of bounds.
Type loops use Copy() proc at the list.

File: code/game/machinery/telecomms/broadcaster.dm (hunk 1 above)
Comment (eswordthecat) [inline code/game/machinery/telecomms/broadcaster.dm]: get_hearers_in_view fetches all atom with FLAG = HEAR

**Pull request merged**